contracts/voting-verifier/src/execute.rs

```diff
@@ -171,21 +230,40 @@ pub fn vote(
 ) -> Result<Response, ContractError> {
     let poll = POLLS
         .may_load(deps.storage, poll_id)?
-        .ok_or(ContractError::PollNotFound)?
-        .try_map(|poll| {
-            poll.cast_vote(env.block.height, &info.sender, votes)
-                .map_err(ContractError::from)
-        })?;
+        .ok_or(ContractError::PollNotFound)?;
 
+    let initial_results = get_indices_with_quorum(&poll);
+
+    let poll = poll.try_map(|poll| {
+        poll.cast_vote(env.block.height, &info.sender, votes)
+            .map_err(ContractError::from)
+    })?;
     POLLS.save(deps.storage, poll_id, &poll)?;
 
-    Ok(Response::new().add_event(
-        Voted {
-            poll_id,
-            voter: info.sender,
-        }
-        .into(),
-    ))
+    let final_results = get_indices_with_quorum(&poll);
+
+    // we only want to emit a quorum reached event when quorum is first reached, and not subsequently
+    let reached_quorum: Vec<&(u32, Vote)> = final_results
+        .iter()
+        .filter(|e| !initial_results.contains(e))
+        .collect();
+
+    let events = reached_quorum
```

Comment:
So `events` would be empty if this is not the first time when the quorum has been reached?

Reply:
yes

Reply:
I don't understand how it's achieved

Reply:
What do you mean? initial_results are the messages that have reached quorum before tallying this vote, final_results are the messages that have reached quorum after tallying this vote, and then reached_quorum is everything that is in final_results but not in initial_results, so that would be the messages for which this particular vote caused them to achieve quorum.

Reply:
How can there be anything in the initial_results that has reached quorum when we are trying to find out if this is the very first time quorum has been reached?